tasks/need_fixing_linters/golangci_lint_parser.py

```diff
@@ -0,0 +1,125 @@
+        if match is None:
+            continue
+
+        file_path, row, col, lint, _, descr, linter = match.groups()
+        full_path = os.path.normpath(os.path.join(current_module, file_path))
+        base_path = full_path.removeprefix(os.getcwd()).removeprefix('/')
+        lints.append((base_path, row, col, lint, descr, linter))
+
+    lints_per_linter = dict()
+    for base_path, row, col, lint, descr, linter in lints:
+        if linter not in lints_per_linter:
+            lints_per_linter[linter] = []
+        lints_per_linter[linter].append((base_path, row, col, lint, descr))
+
+    ## Uncomment to get the number of lints per linter
+    # for linter, lints in lints_per_linter.items():
+    #     print(f"{linter}: {len(lints)}")
+
+    return lints_per_linter
+
+
+def filter_lints(lints_per_linter, filter_team: str=None, filter_linters: str=None):
+    """
+    Keeps only the lints from a specific team and specific linters.
+
+        Parameters:
+            filter_team (str): Keep only the lints from the files owned by the filter_team. None will keep lints from all teams.
+            filter_linters (str): Comma-separated linters to keep. None will keep lints from all linters.
+    """
+    list_filter_linters = filter_linters.split(',') if filter_linters else []
+    filtered_lints = dict()
+    codeowners = read_owners(CODEOWNERS_FILE_PATH)
+    for linter in lints_per_linter:
+
+        # If either we didn't set a filter or the linter is in the filter list
+        if not filter_linters or linter in list_filter_linters:
+            if linter not in filtered_lints:
+                filtered_lints[linter] = []
+
+            if not filter_team:
+                filtered_lints[linter] = lints_per_linter[linter]
+            else:
+                # Filter only the lints owned by the filter_team
+                for lint in lints_per_linter[linter]:
+                    if codeowners.of(lint[0]) == []:
+                        continue
+                    team_owning_the_lint = codeowners.of(lint[0])[0][1]
+                    if team_owning_the_lint == filter_team:
+                        filtered_lints[linter].append(lint)
+    return filtered_lints
+
+
+def display_result(filtered_lints):
+    """
+    Displays results
+    """
+    output = ""
+    for linter in filtered_lints:
+        output+= f"[{linter}]\n"
+        for lint in filtered_lints[linter]:
+            output += f"\n{lint[0]}:{lint[1]}:{lint[2]} {lint[3]}{lint[4]}"
+        output += "\n"
+    return output
+
+def merge_results(results_per_os_x_arch: dict[str: str]):
+    """
+    Merge golangci-lint output
+    """
+    merged_lints_per_linter = {}
+    for os_x_arch, result in results_per_os_x_arch.items():
+        lints_per_linter = parse_file(result)
+        for linter, lints in lints_per_linter.items():
+            if linter not in merged_lints_per_linter:
+                merged_lints_per_linter[linter] = []
```

Comment:
you can use a `defaultdict` from [collections](https://docs.python.org/3/library/collections.html?highlight=defaultdict#collections.defaultdict) to save this block 
```python
merged_lints_per_linter = defaultdict(list)
```
You can also maybe use directly a `defaultdict(set)` to remove duplicates and use merged lints as is not to parse twice the list, wdyt?

Reply:
Didn't know about `defaultdict` ! That's a great feature, thank you ! Fixed in [8a4ad8d](https://github.com/DataDog/datadog-agent/pull/19015/commits/8a4ad8df6bc15855388760fd383eb8ca25948ab0).